components/ca-import.js

```diff
@@ -0,0 +1,727 @@
+            const el = e.target || e.srcElement;
+
+            switch (this.type) {
+
+                case 'csv': el.href = `data:text/plain;charset=utf-8,${encodeURIComponent(this.headers)}`; break;
+
+                case 'xlsx': {
+
+                    cancelEvent(e);
+
+                    const rows = [];      // collection of rows
+                    const rowData = {};   // individual row item (js object key/values)
+
+                     // convert header array into a JSON key/value object
+                    for (let i = 0, l = self.headers.length; i < l; i++) {
+                        rowData[self.headers[i]] = '';
+                    }
+
+                     // get the example row if this.firstRow attribute is set
+                    const exampleRow = this.buildExampleRow.call(this);
+                    if (exampleRow) {
+                        rows.push(exampleRow);
+                    }
+
+                     // as its a template create some blank rows to ensure formatting is respected
+                    let numOfBlankRows = 25;
+                    while (numOfBlankRows--) {
+                        rows.push(rowData);
+                    }
+
+                     // post to server via iframe, server will convert to excel and trigger a download
+                    postViaIframe(this.templateConversionUrl, {
+                        json: JSON.stringify(rows)
+                    });
+
+                } break;
+
+                default: break;
+            }
+        }
+
+         /**
+          * clear the file upload path to allow the same file to trigger the onchange event
+          * @param {event} e, event triggered
+          * @returns {undefined} nothing.
+          */
+        fileClick(e) {
+            this.input.value = '';
+        }
+
+         /**
+          * File event handler
+          * @param {event} e, event triggered
+          * @returns {undefined} example row.
+          */
+        fileHandler(e = event) {
+            const el = e.target || e.srcElement;
+
+            if (el && el.files) {
+
+                const file = el.files[0];
+
+                switch (this.type) {
+                    case 'csv': this.readFileAsCsv.call(this, file); break;
+                    case 'xlsx': this.convertXlsxToJson.call(this); break;
+                    default: break;
+                }
+            }
+        }
+
+         /**
+          * check each row against the schema, if its invalid add it to the list or errors returned
+          * @param {object} dataObjects - array of excel data in object (key/value) format
+          * @returns {array} list of errors.
+          */
+        validateRows(dataObjects) {
+            const errors = [];
+
+            if (this.schema && this.schema.properties) {
+
+                dataObjects.forEach(function(row, index) {
+
+                     // eslint-disable-next-line no-restricted-syntax
+                    for (const key in row) {
```

Comment:
Could be Objects.keys(row).forEach((key) => { - and remove the lint override and the call to hasOwnProperty().